.circleci/config.yml

```diff
@@ -1,56 +1,127 @@
+            git config --global user.email "ci@kubos.com"
+            git config --global user.name "CircleCI"
+            git tag -m "CI tagging test-release with ${new_tag} on $(date '+%m-%d-%y@%H:%M:%S')" ${new_tag}
+            git push origin ${new_tag}
+            # Immediately delete the tag -- we're just checking that we have access
+            git push --delete origin ${new_tag}
 
+  deploy:
+    executor: python/default
+    steps:
+      - add_ssh_keys:
+          # This is the writable deploy key in Github called "Write-Access Key for Publishing and Git Tagging"
+          # The keys are stored in LastPass under CircleCI
+          fingerprints:
+            - "12:3e:11:34:9e:3f:b3:6e:9c:64:f6:5e:1f:d9:80:49"
+      - checkout
+      - python/install-packages:
+          pkg-manager: pip
+          pypi-cache: false
+          args: -r deploy_requirements.txt
+      - python/dist
       - run:
-          name: install dependencies
+          # twine depends on environment variables being defined on the project
+          name: Publish to PyPi
+          command: | 
+            twine upload -u __token__ -p ${PYPI_TOKEN} dist/*.tar.gz dist/*.whl
+
+      - run:    
+          name: Tag release
           command: |
-            python3 -m venv venv
-            . venv/bin/activate
-            pip install -r requirements.txt
+            set -exo pipefail
+            new_tag="v$(cat VERSION)"
+            git config --global user.email "ci@kubos.com"
+            git config --global user.name "CircleCI"
+            git tag -m "CI tagging release with ${new_tag} on $(date '+%m-%d-%y@%H:%M:%S')" ${new_tag}
+            git push origin ${new_tag} 
 
-      - save_cache:
-          paths:
-            - ./venv
-          key: v1-dependencies-{{ checksum "requirements.txt" }}
 
+  # Test multiple python version by running multiple jobs
+  orbTest:
+    parameters:
+      tag:
+        description: "python version"
+        default: "3.8"
+        type: string
+    executor:
+      name: python/default
+      tag: <<parameters.tag>>
+    steps: 
+      - checkout
+      - python/install-packages:
+          pkg-manager: pip
+          pypi-cache: false
+          args: -r test_requirements.txt
       - run:
-          name: run tests
           command: |
-            . venv/bin/activate
             pytest
+          name: Test it
+  
+  # Test multiple python version by running a single job with tox
+  toxTest:
+    working_directory: ~/repo
+    docker:
+      - image: cimg/base:2021.04
+    steps:
+      - checkout
+      - setup_remote_docker:
+          version: 19.03.13
+      - run: ./dockertest.sh
 
-      - store_artifacts:
-          path: test-reports
-          destination: test-reports
 
 workflows:
   version: 2
-  test:
+  test-and-maybe-deploy:
     jobs:
-      - build
+    - toxTest
+    - orbTest:
+        matrix:
+          parameters:
+            tag: ["3.6", "3.7", "3.8", "3.9"]   
```

Comment:
I'm not sure it's a bad thing, but I notice that this list doesn't match the "envlist" in tox.ini.

Reply:
Yeah, I couldn't get 3.9 to work in tox, so I tried another method (multiple jobs). The tox one is easier to run locally, but I think the multiple jobs is better for CI. Not sure how to bridge the two or reduce. Maybe `docker-compose` to run multiple locally?